HCIDOCS-187: Document procedure for changing BMC address after installation (#76624)
openshift/openshift-docs: modules/ipi-install-bmc-addressing.adoc

Most vendors support Baseboard Management Controller (BMC) addressing with the Intelligent Platform Management Interface (IPMI). IPMI does not encrypt communications. It is suitable for use within a data center over a secured or dedicated management network. Check with your vendor to see if they support Redfish network boot. Redfish delivers simple and secure management for converged, hybrid IT and the Software Defined Data Center (SDDC). Redfish is human readable and machine capable, and leverages common internet and web services standards to expose information directly to the modern tool chain. If your hardware does not support Redfish network boot, use IPMI.

> If you need to modify the BMC address or credentials after installation, see the _Additional resources_ section.

Reviewer: This epic applies to BMC address only, not credentials. While your statement is correct in the context of this doc, the following must be made clear about editing the BMC address specifically:

The BMC address can be edited in two ways/cases: with the detached annotation, and when the BMH is still in the 'Registering' state (in BMO). 
[From my observation, a BMH very quickly moves out of the Registering state as soon as the credentials are verified, but if you enter the BMC address incorrectly (by mistake, for example), the BMH will not move past the Registering state.] While the BMH is in the Registering state, the BMC address can be edited in the BMH resource (without the need for detachment). In all other BMH states, you have to use the detach annotation.

Author: @MahnoorAsghar I've made some changes. Can you have a look and give an LGTM if it's okay?

> You can modify the BMC address during installation while the node is in the `Registering` state. If you need to modify the BMC address after the node leaves the `Registering` state, or if you need to modify the BMC credentials, you must disconnect the node from Ironic, edit the `BareMetalHost` resource, and reconnect the node to Ironic. See the _Additional resources_ section for details.

Reviewer 2: Is there a way of creating a unique add resources under this section as the one I think your pointing to is very far down the document. If someone adds an add resource elsewhere, confusion coukld arise.

> You can modify the BMC address during installation while the node is in the `Registering` state. If you need to modify the BMC address after the node leaves the `Registering` state, or if you need to modify the BMC credentials, you must disconnect the node from Ironic, edit the `BareMetalHost` resource, and reconnect the node to Ironic. See the _Additional resources_ section for details.

Reviewer: I highly suspect that you do not need to detach the BMH for editing the BMC credentials.

Reviewer 3: Yes, @MahnoorAsghar is correct. You can change the password any time.
To change the _address_ - effectively pointing the BMH at a completely different host - you must now disconnect from ironic first (previously this was not allowed at all).

          password: <password>
          disableCertificateVerification: True
----

[discrete]
== Redfish APIs

The installation program calls several Redfish API endpoints on the BMC when using installer-provisioned infrastructure on bare metal.

[IMPORTANT]
====
Ensure that your BMC supports all of the Redfish APIs before installation.
====

.Procedure

. Set the BMC user name by running the following command:
+
[source,terminal]
----
$ export USER=<username> <1>
----
<1> Replace `<username>` with the name of the BMC user.

. Set the BMC password by running the following command:
+
[source,terminal]
----
$ export PASS=<password> <1>
----

> <1> Replace `<password>` with the password of the BMC user.

Reviewer 2: I'm not srue these are needed. Customers should be familiar with use-repalced values. If we had a large example, calling out replaceable values would be cumbersome if loads existed.

Reviewer: (Not from the docs team, but FWIW I agree, its unnecessary and increases doc length)

. Set the IP address of the BMC by running the following command:
+
[source,terminal]
----
$ export SERVER=<ip_address> <1>
----
<1> Replace `<ip_address>` with the IP address of the BMC.

. Set the ID of the system by running the following command:
+
[source,terminal]
----
$ export SystemID=<system_id> <1>
----

> <1> Replace `<system_id>` with the system ID. For example, `System.Embedded.1` or `1`. See the following vendor-specific BMC sections for details.

Reviewer 2: This callout is needed as it provides useful input details.

Reviewer: Agreed.

.List of Redfish APIs:

* Check `power on` support by running the following command:
+
[source,terminal]
----
$ curl -u $USER:$PASS -X POST -H'Content-Type: application/json' -H'Accept: application/json' -d '{"ResetType": "On"}' https://$SERVER/redfish/v1/Systems/$SystemID/Actions/ComputerSystem.Reset
----

* Check `power off` support by running the following command:
+
[source,terminal]
----
$ curl -u $USER:$PASS -X POST -H'Content-Type: application/json' -H'Accept: application/json' -d '{"ResetType": "ForceOff"}' https://$SERVER/redfish/v1/Systems/$SystemID/Actions/ComputerSystem.Reset
----

> * Check temporary boot using `pxe` by running the following command:

Reviewer 2: ```suggestion
* Check the temporary boot implementation that uses `pxe` by running the following command:
```

+
[source,terminal]
----
$ curl -u $USER:$PASS -X PATCH -H "Content-Type: application/json"  https://$Server/redfish/v1/Systems/$SystemID/ -d '{"Boot": {"BootSourceOverrideTarget": "pxe", "BootSourceOverrideEnabled": "Once"}}
----

> * Check the ability to set the BIOS boot mode using `Legacy` or `UEFI` by running the following command:

Reviewer 2: ```suggestion
* Check the status of setting the BIOS boot mode that uses `Legacy` or `UEFI` by running the following command:
```

+
[source,terminal]
----
$ curl -u $USER:$PASS -X PATCH -H "Content-Type: application/json"  https://$Server/redfish/v1/Systems/$SystemID/ -d '{"Boot": {"BootSourceOverrideMode":"UEFI"}}
----

.List of Redfish virtual media APIs:

> * Check the ability to set the temporary boot device using `cd` or `dvd` by running the following command:

Reviewer 2: ```suggestion
* Check the ability to set the temporary boot device that uses `cd` or `dvd` by running the following command:
```
